urllib3/connectionpool.py

```diff
@@ -98,6 +98,23 @@ def close(self):
 # This is taken from http://hg.python.org/cpython/file/7aaba721ebc0/Lib/socket.py#l252
 _blocking_errnos = set([errno.EAGAIN, errno.EWOULDBLOCK])
 
+# CPython 2.7 supports an opt-in parameter for response buffering
+# in HTTPConnection.getresponse
+# Rather than checking on each request, we inspect the code object
+# (if available) at import and only pass the argument if it appears
+# in the list of variable names
+def _request_httplib_response_buffering():
```

Comment:
I was trying to avoid that, since `inspect` is a relatively heavy import (not as large as `urllib3` itself, but not small either - on CPython 3.5 it would add 10 net new modules beyond what "import urllib3" already brings in). The theory behind the current approach is that we're *not* trying to detect a documented Python feature - we're trying to detect an undocumented implementation-specific behaviour of CPython 2.7. That means that an attribute error on `im_func` (indicating Python 3's lack of unbound functions), an attribute error on `func_code` (indicating a 2.x implementation without CPython-style code objects), or a lack of `buffering` in `co_varnames` (indicating a method implementation without a `buffering` local variable, and hence definitely no such parameter) all indicate "don't pass `buffering=True`)

You also reminded me that I never followed through on http://bugs.python.org/issue27172 (the `getfullargspec()` deprecation is being reversed, since it now builds on `inspect.signature()` internally, and doesn't lose information the way `getargspec()` does). If the release manager approves, I'll aim to get that addressed for 3.6.0 (and backported to 3.5).